src/main/java/hudson/plugins/gradle/injection/MavenBuildScanInjection.java

```diff
@@ -63,12 +63,22 @@ private void injectMavenExtension(FilePath rootPath) {
             if (getGlobalEnvVar(GE_URL_VAR) != null) {
                 mavenOptsKeyValuePairs.add(asSystemProperty(GRADLE_ENTERPRISE_URL_PROPERTY_KEY, getGlobalEnvVar(GE_URL_VAR)));
             }
-            rootPath.act(new SetMavenOpts(mavenOptsKeyValuePairs));
+            appendMavenOpts(node, mavenOptsKeyValuePairs);
         } catch (IOException | InterruptedException e) {
             throw new IllegalStateException(e);
         }
     }
 
+    void appendMavenOpts(Node node, List<String> mavenOptsKeyValuePairs) throws IOException, InterruptedException {
+        EnvVars nodeEnvVars = EnvVars.getRemote(node.getChannel());
+        String mavenOpts = String.join(" ", mavenOptsKeyValuePairs);
+        String mavenOptsEnvName = "MAVEN_OPTS";
+        if (nodeEnvVars.get(mavenOptsEnvName) != null) {
+            mavenOpts = nodeEnvVars.get(mavenOptsEnvName) + " " + mavenOpts;
```

Comment:
We need to check here whether we've not already appended (e.g. because of another configuration change that happened before)